src/openzaak/components/catalogi/api/viewsets/zaaktype.py

```diff
@@ -80,7 +74,47 @@ class ZaakTypeViewSet(
         "list": SCOPE_ZAAKTYPES_READ,
         "retrieve": SCOPE_ZAAKTYPES_READ,
         "create": SCOPE_ZAAKTYPES_WRITE,
+        "update": SCOPE_ZAAKTYPES_WRITE,
+        "partial_update": SCOPE_ZAAKTYPES_WRITE,
         "destroy": SCOPE_ZAAKTYPES_WRITE,
         "publish": SCOPE_ZAAKTYPES_WRITE,
     }
-    concept_related_fields = ["besluittype_set"]
+    concept_related_fields = ["besluittypen", "informatieobjecttypen"]
+    relation_fields = ["zaaktypenrelaties"]
+
+    def perform_create(self, serializer):
+        for field_name in self.relation_fields:
+            field = serializer.validated_data.get(field_name, [])
+            for related_object in field:
+                # TODO fix url lookup to check if concept
+                if not related_object["gerelateerd_zaaktype"]:
+                    msg = _(
+                        f"Relations to non-concept {field_name} object can't be created"
+                    )
+                    raise PermissionDenied(detail=msg)
```

Comment:
can you please change `PermissionDenied` into `ValidationError`? I think all `PermissionDenied` in context of `concept field should be replaced to `ValidationError`. @sergei-maertens can you confirm (or not confirm ) it? 

I did it for ZTC in reference implementation (PR - https://github.com/VNG-Realisatie/gemma-zaaktypecatalogus/pull/77)

Reply:
This code was for the concept business logic between related zaaktypen (`gerelateerdeZaaktypen`), but I discussed with Joeri and we have decided to not apply this logic to related zaaktypes, because otherwise it would be impossible to publish them (i.e. if I have zaaktype X and zaaktype Y that are related, I can't publish X because Y is still a concept, and I can't publish Y because X is still a concept).

So I will remove this code altogether